Jenkinsfile

```diff
@@ -141,24 +150,30 @@ def storeJunitResults(String path) {
 }
 
 def collectAarMetrics() {
-  sh '''set -xe
-  cd realm/realm-library/build/outputs/aar
-  unzip realm-android-library-release.aar -d unzipped
-  find $ANDROID_HOME -name dx | sort -r | head -n 1 > dx
-  $(cat dx) --dex --output=temp.dex unzipped/classes.jar
-  cat temp.dex | head -c 92 | tail -c 4 | hexdump -e '1/4 "%d"' > methods
-  '''
-
-  sendMetrics('methods', readFile('realm/realm-library/build/outputs/aar/methods'))
-
-  def aarFile = findFiles(glob: 'realm/realm-library/build/outputs/aar/realm-android-library-release.aar')[0]
-  sendMetrics('aar_size', aarFile.length as String)
-
-  def soFiles = findFiles(glob: 'realm/realm-library/build/outputs/aar/unzipped/jni/*/librealm-jni.so')
-  for (int i = 0; i < soFiles.length; i++) {
-      def abiName = soFiles[i].path.tokenize('/')[-2]
-      def libSize = soFiles[i].length as String
-      sendTaggedMetric('abi_size', libSize, 'type', abiName)
+  def flavors = ['base', 'objectServer']
+  for (def i = 0; i < flavors.size(); i++) {
+    def flavor = flavors[i]
```

Comment:
Iterators are non-serializable, which is something you want to avoid in Jenkinsfiles, since Jenkins saves the state to disk in order to be able to recover from master reboots. See here: https://github.com/jenkinsci/pipeline-examples/blob/master/docs/BEST_PRACTICES.md